pkg/reconciliation/reconcile_fql.go

```diff
@@ -1,98 +1,53 @@
 package reconciliation
 
 import (
-	"encoding/json"
 	"errors"
-	"fmt"
-	"strconv"
-	"strings"
 
 	"github.com/k8ssandra/cass-operator/pkg/httphelper"
 	"github.com/k8ssandra/cass-operator/pkg/internal/result"
 )
 
-// parseFQLFromConfig parses the DC config field to determine whether FQL should be enabled based on the presence of full_query_logging_options within the cassandra-yaml field.
-// To ease integration into the reconciliation process, it returns (shouldFQLBeEnabled, serverMajorVersion, ReconcileResult) where the ReconcileResult may be result.Error() or result.Continue().
-func parseFQLFromConfig(rc *ReconciliationContext) (bool, int64, result.ReconcileResult) {
+// CheckFullQueryLogging sets FQL enabled or disabled based on the `enableFQL` parameter, and takes serverMajorVersion for additional validation.
+// It calls the NodeMgmtClient which calls the Cassandra management API and returns a result.ReconcileResult.
+func (rc *ReconciliationContext) CheckFullQueryLogging() result.ReconcileResult {
```

Comment:
My earlier comments RE methods on concrete types apply here. It will be very hard to stub out this method to unit test any callers. Could we consider changing it to `func CheckFullQueryLogging(rc *ReconciliationContext)`?

Reply:
A method alone shouldn't be stubbed in this case, as the test should verify the behavior of ReconciliationContext. If you need to test something else, then make it accept something else than ReconciliationContext as input.

Reply:
I'll agree to disagree here too. I'm not going to get dogmatic about the 'right way' to do testing (if I was even 100% certain of it myself 🙂)